tendrl/node_agent/manager/manager.py

```diff
@@ -85,49 +81,40 @@ def _run(self):
         LOG.info("%s complete" % self.__class__.__name__)
 
 
-class Manager(object):
+class NodeAgentManager(Manager):
     """manage user request thread
 
     """
 
     def __init__(self, machine_id):
         self._complete = gevent.event.Event()
-
-        self._user_request_thread = EtcdThread(self)
-        self._discovery_thread = TopLevelEvents(self)
-        self.persister = Persister()
+        # Initialize the state sync thread which gets the underlying
+        # node details and pushes the same to etcd
+        self._discovery_thread = NodeAgentSyncStateThread(self)
         etcd_kwargs = {'port': int(config.get("common", "etcd_port")),
                        'host': config.get("common", "etcd_connection")}
         self.etcd_client = etcd.Client(**etcd_kwargs)
-
-        self.register_node(machine_id)
-
-    def stop(self):
-        LOG.info("%s stopping" % self.__class__.__name__)
-        self._user_request_thread.stop()
-        self._discovery_thread.stop()
-        self.persister.stop()
-
-    def start(self):
-        LOG.info("%s starting" % self.__class__.__name__)
-        self._user_request_thread.start()
-        self._discovery_thread.start()
-        self.persister.start()
-
-    def join(self):
-        LOG.info("%s joining" % self.__class__.__name__)
-        self._user_request_thread.join()
-        self._discovery_thread.join()
-        self.persister.join()
-
-    def register_node(self, machine_id):
-        local_node_context = utils.get_local_node_context()
+        local_node_context = utils.set_local_node_context()
         if local_node_context:
             if utils.get_node_context(self.etcd_client, local_node_context) \
                     is None:
                 utils.delete_local_node_context()
 
-        self.persister.update_node_context(
+        super(
+            NodeAgentManager,
+            self
+        ).__init__(
+            "node",
+            utils.get_local_node_context(),
+            config,
+            self._discovery_thread,
+            NodeAgentEtcdPersister(config),
+            "/tendrl_definitions_node_agent/data"
+        )
+        self.register_node(machine_id)
+
+    def register_node(self, machine_id):
+        self.persister_thread.update_node_context(
```

Comment:
Its actually not required. But retained the register_node as is. This doesnt have any impact and would return same node_context and will not create new one. 
Discussed with @nnDarshan and kept like this.